src/pwck.c

```diff
@@ -816,7 +816,7 @@ static void check_spw_file (int *errors, bool *changed)
 		if (!quiet) {
 			time_t t = time (NULL);
 			if (   (t != 0)
-			    && (spw->sp_lstchg > (long) t / DAY)) {
+			    && (spw->sp_lstchg > (long) (t / DAY))) {
```

Comment:
I think this patch should be dropped, right?  <https://github.com/shadow-maint/shadow/pull/876/commits/17f48ff54675924580da37cb7712df0fdd624d35>

Reply:
Or maybe the commit message changed.

Reply:
The cast has to disappear, otherwise `t` is capped at LONG_MAX for 32 bit systems with a 64 bit time_t. Adjusted.